lib/node_modules/@stdlib/blas/base/sspmv/lib/ndarray.js

```diff
@@ -0,0 +1,162 @@
+
+var sfill = require( '@stdlib/blas/ext/base/sfill' );
+var sscal = require( '@stdlib/blas/base/sscal' );
+var f32 = require( '@stdlib/number/float64/base/to-float32' );
+var isLayout = require( '@stdlib/blas/base/assert/is-layout' );
+var isMatrixTriangle = require( '@stdlib/blas/base/assert/is-matrix-triangle' );
+
+
+// MAIN //
+
+/**
+* Performs the matrix-vector operation `y = alpha*A*x + beta*y` where `alpha` and `beta` are scalars, `x` and `y` are `N` element vectors, and `A` is an `N` by `N` symmetric matrix, supplied in packed form.
+*
+* @param {string} order - storage layout
+* @param {string} uplo - specifies whether the upper or lower triangular part of the symmetric matrix `A` should be referenced
+* @param {NonNegativeInteger} N - number of elements along each dimension of `A`
+* @param {number} alpha - scalar constant
+* @param {Float32Array} AP - packed form of symmetric matrix `A`
+* @param {Float32Array} x - first input array
+* @param {integer} strideX - `x` stride length
+* @param {NonNegativeInteger} offsetX - `x` index offset
+* @param {number} beta - scalar constant
+* @param {Float32Array} y - second input array
+* @param {integer} strideY - `y` stride length
+* @param {NonNegativeInteger} offsetY - `y` index offset
+* @throws {TypeError} first argument must be a valid order
+* @throws {TypeError} second argument must specify whether to reference the lower or upper triangular matrix
+* @throws {RangeError} third argument must be a nonnegative integer
+* @throws {RangeError} seventh argument must be non-zero
+* @throws {RangeError} eleventh argument must be non-zero
+* @returns {Float32Array} `y`
+*
+* @example
+* var Float32Array = require( '@stdlib/array/float32' );
+*
+* var A = new Float32Array( [ 1.0, 2.0, 3.0, 4.0, 5.0, 6.0 ] );
+* var x = new Float32Array( [ 1.0, 1.0, 1.0 ] );
+* var y = new Float32Array( [ 1.0, 1.0, 1.0 ] );
+*
+* sspmv( 'column-major', 'lower', 3, 1.0, A, x, 1, 0, 1.0, y, 1, 0 );
+* // y => <Float32Array>[ ~7.0, ~12.0, ~15.0 ]
+*/
+function sspmv( order, uplo, N, alpha, AP, x, strideX, offsetX, beta, y, strideY, offsetY ) { // eslint-disable-line max-params
+	var temp1;
+	var temp2;
+	var ix;
+	var iy;
+	var jx;
+	var jy;
+	var kk;
+	var kx;
+	var ky;
+	var sy;
+	var j;
+	var k;
+
+	if ( !isLayout( order ) ) {
+		throw new TypeError( 'invalid argument. First argument must be a valid order. Value: `%s`.', order );
+	}
+	if ( !isMatrixTriangle( uplo ) ) {
+		throw new TypeError( 'invalid argument. Second argument must specify whether to reference the lower or upper triangular matrix. Value: `%s`.', uplo );
+	}
+	if ( N < 0 ) {
+		throw new RangeError( 'invalid argument. Third argument must be a nonnegative integer. Value: `%d`.', N );
+	}
+	if ( strideX === 0 ) {
+		throw new RangeError( 'invalid argument. Seventh argument must be non-zero. Value: `%d`.', strideX );
+	}
+	if ( strideY === 0 ) {
+		throw new RangeError( 'invalid argument. Tenth argument must be non-zero. Value: `%d`.', strideY );
```

Comment:
```suggestion
		throw new RangeError( 'invalid argument. Eleventh argument must be non-zero. Value: `%d`.', strideY );
```

You'll want to make sure in the error messages that the argument number is correct. This will differ between the main and ndarray interfaces.